Modernize project structure

julia-sprenger opened the pull request:
- switch to pyproject.toml only (no `setup.py`)
- separate test suite from package code
- introduce `src` folder to host package code

TODO:
- [ ] further file cleanup (Are ABSTRACT, MANIFEST needed?)
- [ ] extensive testing @brown170 Could you check if this version can be installed on your mac?

brown170 commented:
@julia-sprenger:

- run-tests.sh hasn't been needed for a while (that was for a very old CI/CD system)
- ABSTRACT, MANIFEST were (are?) a LLNL requirement.  LLNL did not specify where we store them though.  So, if you can think of a better place (doc/LLNL/?), go for it

brown170 commented:
@julia-sprenger I can verify that I can install this and run the tests.  However when I tried to install the EXFOR databases, I noted that the filenames for the CLI changed.  This is OK, but the first script (install-exfor-db) calls the second (install-exfor-db-index) and the filenames need to be updated.  Along with that, the README needs updates.

brown170 commented:
@julia-sprenger the ABSTRACT is a requirement from LLNL, but maybe it can be hidden in the doc/ directory.   The MANIFEST file is there for the old setup.py based installations.  I think it can be removed.

julia-sprenger commented:
Hi @brown170!

I removed the Manifest and updated the Readme.
Regarding the ABSTRACT.in and the current status of the documentation it would make sense to switch to a more comprehensive documentation framework like sphinx and host the documentation somewhere (e.g. readthedocs). I am not going to propose those change in this PR since it's a new chapter of modernization and this PR  here is already quite extensive.

Does that make sense to you?

brown170 commented:
@julia-sprenger I agree!
